[ BOL-106] 임시 멤버 추가 기능 & 검색 없을 때 기능 기능 구현

prk4224 opened the pull request:
### Reference
- [Jira Ticket](https://yapp22-aos2.atlassian.net/browse/BOL-106)

### 참고 사항

P1: 꼭 반영해주세요 (Request changes)
P2: 적극적으로 고려해주세요 (Request changes)
P3: 웬만하면 반영해 주세요 (Comment)
P4: 반영해도 좋고 넘어가도 좋습니다 (Approve)
P5: 그냥 사소한 의견입니다 (Approve)

prk4224 committed:
feat: 임시멤버 추가 다이얼 로그 화면 구현

prk4224 committed:
Merge branch 'feat/BOL-53' into feat/BOL-106

prk4224 committed:
refactor:  코드 클린 업

prk4224 committed:
feat: 임시 멤버 추가 다이얼 로그에 이미지 추가

prk4224 committed:
feat: 임시 멤버 닉네임 중복 체크 API 연결

prk4224 committed:
faet: 멤버 검색 결과 없을 때 기능 구현

prk4224 committed:
Merge branch 'develop' into feat/BOL-106

# Conflicts:
#	app/src/main/java/com/yapp/bol/app/di/DataSourceModule.kt
#	app/src/main/java/com/yapp/bol/app/di/NetworkModule.kt
#	app/src/main/java/com/yapp/bol/app/di/RepositoryModule.kt
#	buildSrc/src/main/java/com/yapp/bol/Dependencies.kt
#	data/src/main/java/com/yapp/bol/data/datasource/RemoteDataSource.kt
#	data/src/main/java/com/yapp/bol/data/datasource/impl/RemoteDataSourceImpl.kt
#	data/src/main/java/com/yapp/bol/data/mapper/MapperToDomain.kt
#	data/src/main/java/com/yapp/bol/data/remote/GroupApi.kt
#	data/src/main/java/com/yapp/bol/data/repository/RepositoryImpl.kt
#	domain/src/main/java/com/yapp/bol/domain/repository/Repository.kt
#	presentation/src/main/AndroidManifest.xml
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/MatchActivity.kt
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/MatchViewModel.kt
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/game_select/GameSelectAdapter.kt
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/game_select/GameSelectFragment.kt
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/member_select/MemberSelectFragment.kt
#	presentation/src/main/java/com/yapp/bol/presentation/viewmodel/login/LoginViewModel.kt
#	presentation/src/main/res/layout/activity_match.xml
#	presentation/src/main/res/layout/fragment_game_selete.xml
#	presentation/src/main/res/layout/fragment_member_select.xml
#	presentation/src/main/res/values/colors.xml
#	presentation/src/main/res/values/strings.xml

prk4224 committed:
refactor: Mock Prefix 제거

prk4224 committed:
Merge branch 'feat/BOL-53' into feat/BOL-106

# Conflicts:
#	app/src/main/java/com/yapp/bol/app/di/RepositoryModule.kt
#	buildSrc/src/main/java/com/yapp/bol/Dependencies.kt
#	data/src/main/java/com/yapp/bol/data/datasource/RemoteDataSource.kt
#	data/src/main/java/com/yapp/bol/data/datasource/impl/RemoteDataSourceImpl.kt
#	data/src/main/java/com/yapp/bol/data/mapper/MapperToDomain.kt
#	data/src/main/java/com/yapp/bol/data/remote/GroupApi.kt
#	data/src/main/java/com/yapp/bol/data/repository/RepositoryImpl.kt
#	domain/src/main/java/com/yapp/bol/domain/repository/Repository.kt
#	presentation/src/main/java/com/yapp/bol/presentation/view/match/member_select/MemberSelectFragment.kt
#	presentation/src/main/res/values/strings.xml

prk4224 committed:
Merge branch 'feat/BOL-53' into feat/BOL-106

prk4224 committed:
Merge branch 'feat/BOL-53' into feat/BOL-106

prk4224 committed:
refactor: ktlint 수정

prk4224 committed:
refactor: Query import 추가

eunjjungg commented on presentation/src/main/java/com/yapp/bol/presentation/view/match/MatchViewModel.kt:
```diff
@@ -32,20 +32,39 @@ class MatchViewModel @Inject constructor(
     val gameList = _gameList
 
+    private val _isNickNameValidate = MutableLiveData(false)
+    val isNickNameValidate = _isNickNameValidate
```
p4: 요거 이렇게 하면 `gameList`, `isNickNameValidate`에 대해 외부에서 수정이 가능한 것 같습니다..! 혹시 의도하신 거라면 넘겨주세요~! 아래는 제가 테스트했던 코드입니다!

``` kotlin
private val _a: MutableLiveData<Int> = MutableLiveData(3)
val a = _a

override fun onViewCreatedAction() {
    super.onViewCreatedAction()
    a.value = 10
    Log.d("mutable check?", a.value.toString()) // 10 출력 
}
```

eunjjungg commented on presentation/src/main/java/com/yapp/bol/presentation/utils/GuestAddDialog.kt:
```diff
@@ -0,0 +1,86 @@
+        const val FIRST_COLOR_CHANGE_START_POINT = 3
+        const val FIRST_COLOR_CHANGE_END_POINT = 18
+        const val SECOND_COLOR_CHANGE_START_POINT = 24
+        const val SECOND_COLOR_CHANGE_END_POINT = 29
```
p4: 혹시 이 부분에 대해서 이 클래스 안에서만 사용하실 거라면 가시성을 private으로 변경해주셔도 좋을 것 같습니다 ㅎㅎ 🫠 그리고 `PROFILE_NAME_MAX_LENGTH` 상수는 다른 파일에서 하나 더 선언해두셨더라구요..! Dialog들에서 사용하실 거라면 Config로 빼도 좋을 것 같다는 생각도 들었습니다~! 혹시 사유가 있으셔서 그러신거라면 넘겨주세요!! 🫡

eunjjungg commented on presentation/src/main/java/com/yapp/bol/presentation/utils/GuestAddDialog.kt:
```diff
@@ -0,0 +1,86 @@
+    private fun setGuestAddDescription() {
+
+        val description = context.resources.getString(R.string.guest_description_text)
+        val spannableString = SpannableString(description)
```
p5: SpannableString의 존재를 처음 알았어요..!!! 이렇게 텍스트에서 색상을 바꿀 수 있군요..! 하핳 감사합니다!! 👍

eunjjungg reviewed (commented):
고생하셨습니다!~~ 👍 많이 배워갑니다!! ㅎㅎ

ckrudals commented on presentation/src/main/java/com/yapp/bol/presentation/view/match/MatchViewModel.kt:
```diff
@@ -32,20 +32,39 @@ class MatchViewModel @Inject constructor(
+        initialized()
+    }
+
+    private fun initialized() {
```
p3 : 함수의 이름은 동사로 짓는게 규칙이므로 `init()` 으로 짓는게 어떤가요?

ckrudals commented on presentation/src/main/java/com/yapp/bol/presentation/view/match/MatchViewModel.kt:
```diff
@@ -32,20 +32,39 @@ class MatchViewModel @Inject constructor(
     }
 
-    private fun getGameList() {
+    fun getValidateNickName(groupId: Int, nickname: String) {
```
p3 : `checkValidNickname` 요정도 이름이 적당한 것 같아요

ckrudals commented on presentation/src/main/java/com/yapp/bol/presentation/view/match/member_select/MemberSelectFragment.kt:
```diff
@@ -85,6 +97,22 @@ class MemberSelectFragment : Fragment() {
+            text = searchResult
+            visibility = visible
+        }
     }
```
```suggestion
    private fun setSearchResultNothing(isVisible: Boolean, keyword: String) {
        val searchResult = String.format(resources.getString(R.string.search_result_nothing), keyword)
        binding.viewSearchResultNothing.isVisible = isVisible
        binding.tvSearchResultNothingGuide.isVisible = isVisible
        binding.btnGuestAddNothing.isVisible = isVisible
        binding.tvSearchResultNothing.apply {
            text = searchResult
            this.isVisible = isVisible
        }
    }
```
`isVisible` 을 사용하면 따로 프로퍼티를 안만들어도 될 것 같아요!

ckrudals commented on presentation/src/main/java/com/yapp/bol/presentation/view/match/member_select/ValidateCallBack.kt:
```diff
@@ -0,0 +1,5 @@
+package com.yapp.bol.presentation.view.match.member_select
+
+interface ValidateCallBack {
+    fun setNicknameValid(value: Boolean)
```
p3 : `checkValidNickname` 정도 이이름이면 될것 같아요.

prk4224 committed:
[BOL-106-1] 기능 피드백 수정 (#45)

* refactor: Fragment 애서 사용하는 ViewModel 분리

뒤로가기 등에서 데어트를 지우기 위함.

* feat: 피드백 수정

- 뒤로가기 시 데이터 초기화
- X 버튼 눌렀을 때 키보는 hide 제거
- 멤버 선택시 전체 클릭해도 선택 기능 추가
- 임시멤버 추가 시 키보드 자동 show 기능 추가
- 임시 멤버 이름 입력 뒤 뒤로가기 시 데이터 초기화
- 키보드에 검색 아이콘으로 변경

* refactor: LiveDate Type 추가

* fix: 키도브 Hide 시 dialog가 Show 상태일때문 내리는 로직 수정

* refactpr: multipart Mime Type으로 변경

* refactor: 코드 스타일 수정 및 불필요한 기능 제거

* feat: 디자인 추가

* refactor: 검색결과 없을 때 임시 멤버 클릭 시 키보드 내리기

* refactor: ktlint 수정

* reafactor: ktlint 수정

* [BOL-64] 게임 결과 기록 화면 구현 (#49)

* feat: Image file 생성

* feat: 게임 기록 화면 구현 및 네비게이션 연결

* feat: 게임 결과 기록 Fragment 생성 및 ViewModel 생성

입력 점수에 따른 자동 순위 변경 기능 구현

* refactor: 게임 리스트 불러오는 Url 수정

* refactor: 멤버 검색 결과 없을 때 + 버튼 활성화 상태 체크

* feat: MemberResultItem 생성

* fix: RecyclerView EditText 중복 현상 해결

* feat: 필요한 icon 추가

* feat: 결과 기록 화면 구현

* feat: 기록을 전부 하면 기록 완료 버튼 활성화

* feat: 동점자 처리 기능 추가

* feat: 1위 Text Color 변경

* feat: 필요한 string & 컬러 추가 및 게임 결과 다이얼 로구 구현

* feat: progressbar 구현 및 XML 코드 refactoring

* refacter: Code Refactoring

* feat: Bundle 을 이용해서 값을 넘겨주기 위해 MemberItem 에  Parcelable 상속

* feat:   다이얼 로크 크기 변경하는 로직에서 ResultDialog 일때 고정값 사용하는 로직으로 수정

* retactor: player 순서 변경에서 불필요한 로직 수정

* feat: 사용성 편의를 위한 로직 수정 및 현재 시간

- 키보드 내렸을 때, 다음 버튼 눌렀을 때 순위 자동 수정
- 다음 버튼 클릭 시 스크롤 자동 이동
- 기록 완료 버튼 클릭 시 다이얼로그 띄우기
- 아이템 클릭시 EditText 로 자동 focusing

* refactor: Code Refactoring

* refactor: MOVE_SCROLL_POINT 변경

* ktlint 수정

* refacter: 사용하지 않는 params _ 처리

* refactor: ktlint 수정
